src/types/widget.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  Helper,
+  HelperState,
+  SearchResults,
+  InstantSearch,
+  SearchParameters,
+} from './instantsearch';
+
+interface WidgetArgs {
+  helper: Helper;
+  state?: HelperState;
+  results?: SearchResults;
+  instantSearchInstance: InstantSearch;
+}
+
+export interface Widget {
+  init(options: WidgetArgs): void;
+  render(options: WidgetArgs): void;
+  dispose(options: any): SearchParameters;
```

Comment:
We consider the `dispose` method required?

Reply:
We can reiterate on this widget signature. @tkrugg mentioned working on copying this from the one we've got in Angular InstantSearch.